calcom/cal.com
# fix: rescheduling round robin events (emails and calendar events)

## What does this PR do?

Fixes issues with rescheduling round-robin events when they are assigned to a new host. 

1. Emails 
- Old rr host will get cancelation email 
- New rr host will get scheduling emails 
- Other fixed hosts and attendees that stay the same receive rescheduling email

2. Calendar events
- If the organizer of the event changes: Old event gets deleted and a new event gets created with adapted event title 
- If only rr host changed but there is a fixed host as the organizer event just gets rescheduled 

4. Organizer's default location
- If location of event type is oragnizer's default location, it will take the default location of the new rr host

Also fixes that Google Meet link wasn't saved in metadata (videoCallUrl)

Fixes #12350
Fixes #11547
Fixes #12341

 Loom Video: https://www.loom.com/share/e8964cbd58d44877a138c9c0b4a9f4e0

## Type of change


## How should this be tested?

- Create round robin event and assign user1 and user2 as round-robin hosts. 
- Set location as 'organizer's default'
- Make user2's default location Google Meet 
- Connect calendars for both users
- Book rr 
- Check if calendar event is correctly created 
- Reschedule booking (make sure the other user is available, so it will be assigned to a new user)
- Check if calendar event is removed from original booking and new event is created on the new organizer's calendar 
- Check Emails: 
  - previous host -> cancellation email 
  - new host -> scheduling email 
  - attendee -> rescheduling email

## Mandatory Tasks
